rotkehlchen/chain/ethereum/modules/balancer/decoder.py

```diff
@@ -249,14 +249,134 @@ def _maybe_enrich_balancer_v1_events(
                     event.event_type == HistoryEventType.SPEND and
                     event.event_subtype == HistoryEventSubType.NONE
                 ):
-                    event.event_type = HistoryEventType.STAKING
                     event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
                     event.counterparty = CPT_BALANCER_V1
                     event.notes = f'Return {event.balance.amount} {token.symbol} to a Balancer v1 pool'  # noqa: E501
                     return True
 
         return False
 
+    def _check_refunds_v1(
+            self,
+            transaction: EvmTransaction,  # pylint: disable=unused-argument
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+    ) -> list[HistoryBaseEntry]:
+        """
+        It can happen that after sending tokens to the DSProxy in balancer V1 the amount of tokens
+        required for the deposit is lower than the amount sent and then those tokens are returned
+        to the DSProxy and then to the user.
+        """
+        deposited_assets = set()
+
+        for event in decoded_events:
+            if (
+                event.counterparty == CPT_BALANCER_V1 and
+                event.event_type == HistoryEventType.DEPOSIT and
+                event.event_subtype == HistoryEventSubType.DEPOSIT_ASSET
+            ):
+                deposited_assets.add(event.asset)
+            elif (
+                event.counterparty == CPT_BALANCER_V1 and
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.RECEIVE_WRAPPED and
+                event.asset in deposited_assets
+            ):
+                # in this case we got refunded one of the assets deposited
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                asset = event.asset.resolve_to_asset_with_symbol()
+                event.notes = f'Refunded {event.balance.amount} {asset.symbol} after depositing in Balancer V1 pool'  # noqa: E501
+
+        return decoded_events
+
+    def _check_deposits_withdrawals_v1(
+            self,
+            transaction: EvmTransaction,  # pylint: disable=unused-argument
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+    ) -> list[HistoryBaseEntry]:
+        """
+        Check for accounting in v1 that the deposits/withdrawals events have the needed information
+        to process them during accounting.
+        """
+        receive_event, return_event = None, None
```

Comment:
add constraing for the type of transaction this should run for. Otherwise this will run as post decoding for all transactions

Reply:
There is no constraint because the balancer v1 events execute against a DSProxy for each user. I'm not sure if there is any other constraint that can be used as filter. What I do is exit early if there is no balancer v1 events